src/basic/macro.h

```diff
@@ -84,6 +84,14 @@
 #define _variable_no_sanitize_address_
 #endif
 
+/* Apparently there's no has_feature() call defined to check for ubsan, hence let's define this
+ * unconditionally on llvm */
+#if defined(__clang__)
+#define _function_no_sanitize_float_cast_overflow_ __attribute__((no_sanitize("signed-integer-overflow")))
```

Comment:
It seems it should be "float-cast-overflow":
```suggestion
#define _function_no_sanitize_float_cast_overflow_ __attribute__((no_sanitize("float-cast-overflow")))
```

@poettering could you cherry-pick https://github.com/systemd/systemd/pull/15908/commits/67cd35ab0f15f3a4fe3db826674cf08d76387d9a so that the CI could automatically make sure the issue is gone?

Reply:
FWIW to judge from https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78204 gcc has supported `__attribute__((no_sanitize...)` since gcc8 was released. Though given that `float-cast-overflow` is off by default there I probably wouldn't worry about it.